core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1166,7 +1166,14 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
           SqlKind.OTHER_FUNCTION,
           ReturnTypes.BIGINT_NULLABLE,
           null,
-          OperandTypes.NUMERIC.or(OperandTypes.BINARY),
+          OperandTypes.or(OperandTypes.NUMERIC,
+              OperandTypes.BINARY,
+              OperandTypes.BOOLEAN,
+              OperandTypes.CHARACTER,
+              OperandTypes.DATETIME,
```

Comment:
you have here more types than the JIRA title promises.
Also, it seems from the JIRA comments that you are not really supporting these other types yet.

The solution for the question you asked in the JIRA is to pass some type information to the runtime. If the input is a Date, you need to pass a "DATE" flag to the runtime so that it interprets the integer as a date.

Reply:
Thanks for your feedback @mihaibudiu, is there any example I can refer to? I am not familiar with the Calcite runtime and I will take some time to learn it.

Reply:
This function you wrote `bitCountMySQL` is part of the runtime.
You just need to pass it an extra argument that describes the type.

Reply:
Thanks for your guidance, I will try this solution.